### ISSUE TITLE
Task due on Sunday does not appear on Sunday

### ISSUE THREAD

**Issue by deanforrest** [nextcloud/calendar#3003]
### Steps to reproduce
1. Create a task in the tasks app, change the due date to Sunday, and make sure "All Day" is checked.
2. Change the calendar to week view, and go to the week where the task is due.
3. You will notice that the task is not present on Sunday.

### Expected behaviour
The task appears right below the "Sunday" header in week and day view.

### Actual behaviour
The task will only appear in the month view or task view. It will not appear in the week or day view. Note: The calendar will briefly show the task when in week view right before the page is fully loaded, but right after the page and all other calendars are loaded, the Sunday tasks will disappear.

### Calendar app
**Calendar app version:** 2.2.0

**CalDAV-clients used:** iOS CalDAV integration

### Client configuration
**Browser:** Firefox 87 (also present in chromium based browsers like Edge)

**Operating system:** Windows 10

### Server configuration

**Operating system**: Ubuntu 18.04

**Web server:** Apache

**Database:** MariaDB

**PHP version:** 7.4

**Nextcloud Version:** 20.0.8

**Comment by simonvanderveldt** [comment]
Isn't this is generic issue with the week-view? Tasks don't really show up there at all if they only have a due date. It seems like they are actually visible in the weekly view, but it's a super tiny line that represents the task.

**Comment by deanforrest** [comment]
> 
> 
> Isn't this is generic issue with the week-view? Tasks don't really show up there at all if they only have a due date. It seems like they are actually visible in the weekly view, but it's a super tiny line that represents the task.

You are correct with that behavior, only when you set a specific due date and time for the task. If you check "All Day" in the task creation pane, the task will appear right below the date, separate from the timeline. This behavior occurs for all days of the week, unless you set the due date to Sunday, with "All Day" checked. Then the task will not appear under Sunday, with either the week or day view.

**Comment by simonvanderveldt** [comment]
OK, that's indeed something different. Thanks for the clarification.

**Comment by bensuchy** [comment]
I have noticed this as well. Adding some additional behavior that I am seeing:

1. Open Nextcloud Calendar.
2. Navigate to a week where there are tasks due on Sunday.
3. Tasks will briefly appear. Tasks will then disappear one-by-one.
4. Navigate to a different week where there are tasks due on Sunday.
5. Tasks will briefly appear. Tasks will then disappear one-by-one.
6. Navigate back to the original week with tasks due on Sunday.
7. Tasks will NOT briefly appear at all.
8. Navigate to the second week with tasks due on Sunday.
9. Tasks will NOT briefly appear at all.

Essentially, Nextcloud Calendar seems to temporarily remember which tasks/days it has already hidden.

These actions will reset this memory:

- Closing the browser tab and re-opening a new Nextcloud browser tab
- Opening a different Nextcloud app within the same browser tab and then going back to Calendar

These actions do not reset this memory:

- Page refresh
- Hard page refresh (Ctrl+F5)
